src/main/java/ledger/user_interface/ui_controllers/TransactionTableView.java

```diff
@@ -289,7 +292,40 @@ public void updateTransactionTableView() {
         }
         ObservableList<TransactionModel> observableTransactionModels = FXCollections.observableList(models);
 
-        this.setItems(observableTransactionModels);
+//        this.setItems(observableTransactionModels);
+
+        // 1. Wrap the ObservableList in a FilteredList (initially display all data).
+        FilteredList<TransactionModel> filteredData = new FilteredList<>(observableTransactionModels, p -> true);
+
+        // 2. Set the filter Predicate.
+        filteredData.setPredicate(transactionModel -> {
+            // If filter text is empty, display all persons.
+            if (searchFilterString == null || searchFilterString.isEmpty()) {
+                return true;
+            }
+
+            // Compare first name and last name of every person with filter text.
+            String lowerCaseFilter = searchFilterString.toLowerCase();
+
+            if (transactionModel.getAmount().toLowerCase().contains(lowerCaseFilter)) {
```

Comment:
It doesn't matter, I believe. I wrote it like that for consistency's sake, and also to account for the super off-chance that there are special lowercase-able symbols in the amount. Since we're lowercasing the filter string, we should likewise lowercase everything we compare it to.